packages/neos-ui/src/Containers/Modals/SelectNodeType/nodeTypeFilter.js

```diff
@@ -0,0 +1,63 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import TextInput from '@neos-project/react-ui-components/src/TextInput/';
+import IconButton from '@neos-project/react-ui-components/src/IconButton/';
+import Icon from '@neos-project/react-ui-components/src/Icon/';
+import {neos} from '@neos-project/neos-ui-decorators';
+import style from './style.css';
+
+@neos(globalRegistry => ({
+    i18nRegistry: globalRegistry.get('i18n')
+}))
+class NodeTypeFilter extends PureComponent {
+    static propTypes = {
+        onChange: PropTypes.func.isRequired,
+        i18nRegistry: PropTypes.object.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {filterSearchTerm: ''};
```

Comment:
Why do you duplicate the `filterSearchTerm` state here? Just pass it down via the props...